app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt

```diff
@@ -129,10 +144,23 @@ class NewConversationViewModel
     private suspend fun searchPublic(searchTerm: String) {
         publicContactsSearchResult = ContactSearchResult.ExternalContact(SearchResultState.InProgress)
 
-        val result = withContext(Dispatchers.IO) {
+        val result: Result
+        val isFederationSearch = searchTerm.matches(FEDERATION_REGEX)
+        val query: String
+        val domain: String
+
+        if (isFederationSearch) {
+            query = searchTerm.split(VALUE_DOMAIN_SEPARATOR)[0]
+            domain = searchTerm.split(VALUE_DOMAIN_SEPARATOR)[1]
+        } else {
+            query = searchTerm
+            domain = self?.id?.domain ?: ""
+        }
+
+        result = withContext(Dispatchers.IO) {
```

Comment:
no need to assign this result as a local variable, also we have a DispatcherProvider class that you can inject into this VIewModel

Reply:
well actually in this case `result` is used later on

Reply:
true , but from what I see you only assign it here 

```
result = withContext(Dispatchers.IO) 
```

so you could do it like this : 


```
val result = withContext(Dispatchers.IO) 
```